src/managers/auth.py

```diff
@@ -72,7 +72,7 @@ def add_user(self, username: str, password: str, replace: bool = False) -> None:
             return logger.info(f"User {username} already exists, skipping creation")
 
         user_credentials = json.loads(
-            self.workload.exec(command=f"karapace_mkpasswd -u {username} -a sha512 {password}")
+            self.workload.exec(command=f"{SNAP_NAME}.mkpasswd -u {username} -a sha512 {password}")
```

Comment:
**todo** Why not exposing a mkpasswd command in the workload, a bit like I was proposing [here](https://github.com/canonical/karapace-operator/pull/1#discussion_r1571008896). This way the same applies here that we could use just `self.workload.mkpasswd`, that would work both in VM and K8s. Actually, it could be even better to defined a method in the baseclass

```
def mkpasswd(username: str, password: str) -> str
```

and here just use

```
user_credentials = json.loads(self.workload.mkpsswd(username, password))
```

which is substrate independent